src/client/Client.cc

```diff
@@ -5755,13 +5753,45 @@ void Client::handle_cap_grant(MetaSession *session, Inode *in, Cap *cap, const M
     _try_to_trim_inode(in, true);
 }
 
+int Client::mds_check_access(Inode *in, const UserPerm& perms, int mask)
+{
+  const gid_t *gids;
+  int count = perms.get_gids(&gids);
+  std::vector<uint64_t> gid_list;
+  filepath path;
+
+  in->make_long_path(path);
+  if (count) {
+    gid_list.reserve(count);
+    for (int i = 0; i < count; ++i) {
+      gid_list.push_back(gids[i]);
+    }
+  }
+
+  for (auto s: cap_auths) {
+    if (s.match.match(path.get_path(), perms.uid(),perms.gid(), &gid_list)) {
+      if ((mask & MAY_WRITE) && s.writeable &&
+	  s.match.root_squash && ((perms.uid() == 0) || (perms.gid() == 0))) {
```

Comment:
+1

Let's log a message here for debugging.

Reply:
Make sense.